### PR TITLE
Interactivity API docs: Add wp-async directives doc

### PR DESCRIPTION
**Pull request by cbravobernal**
## What?

Add documentation for `data-wp-async` directives.

Completes #61885

### CODE REVIEW

File: docs/reference-guides/interactivity-api/api-reference.md [hunk 1]
```diff
@@ -325,6 +325,11 @@ The `wp-on` directive is executed each time the associated event is triggered.
 
 The callback passed as the reference receives [the event](https://developer.mozilla.org/en-US/docs/Web/API/Event) (`event`), and the returned value by this callback is ignored.
 
+### `wp-on-async`
+
+This directive is a more performant approach of `wp-on`. The action will be yielded to the main thread to not block it, allowing other interactions that otherwise would be waiting on the main thread
+to run sooner. Use this directive for any `wp-on` that doesn't need the `event` object (`event.preventDefault()`, `event.stopPropagation()`, etc.)
```
Comment (westonruter) [inline docs/reference-guides/interactivity-api/api-reference.md]:
```suggestion
to run sooner. Use this directive for any `wp-on` that doesn't need synchronous access to the `event` object, in particular the methods `event.preventDefault()`, `event.stopPropagation()`, and `event.stopImmediatePropagation()`.
```

File: docs/reference-guides/interactivity-api/api-reference.md [hunk 2]
```diff
@@ -354,6 +359,10 @@ store( "myPlugin", {
 
 The callback passed as the reference receives [the event](https://developer.mozilla.org/en-US/docs/Web/API/Event) (`event`), and the returned value by this callback is ignored. When the element is removed from the DOM, the event listener is also removed.
 
+### `wp-on-window-async`
+
+Similar to `wp-on-async`. An optimized version of `wp-on-window` that prevents blocking the main thread on long tasks. Use this directive for any `wp-on-window` that doesn't need the `event` object (`event.preventDefault()`, `event.stopPropagation()`, etc.)
```
Comment (westonruter) [inline docs/reference-guides/interactivity-api/api-reference.md]:
```suggestion
Similar to `wp-on-async`. An optimized version of `wp-on-window` that prevents blocking the main thread on long tasks. Use this directive for any `wp-on` that doesn't need synchronous access to the `event` object, in particular the methods `event.preventDefault()`, `event.stopPropagation()`, and `event.stopImmediatePropagation()`. This event listener is also added as `passive`.
```

File: docs/reference-guides/interactivity-api/api-reference.md [hunk 3]
```diff
@@ -383,6 +392,10 @@ store( "myPlugin", {
 
 The callback passed as the reference receives [the event](https://developer.mozilla.org/en-US/docs/Web/API/Event) (`event`), and the returned value by this callback is ignored. When the element is removed from the DOM, the event listener is also removed.
 
+### `wp-on-document-async`
+
+Similar to `wp-on-async`. An optimized version of `wp-on-document` that prevents blocking the main thread on long tasks. Use this directive for any `wp-on-document` that doesn't need the `event` object (`event.preventDefault()`, `event.stopPropagation()`, etc.)
```
Comment (westonruter) [inline docs/reference-guides/interactivity-api/api-reference.md]:
```suggestion
Similar to `wp-on-async`. An optimized version of `wp-on-document` that prevents blocking the main thread on long tasks. Use this directive for any `wp-on` that doesn't need synchronous access to the `event` object, in particular the methods `event.preventDefault()`, `event.stopPropagation()`, and `event.stopImmediatePropagation()`. This event listener is also added as `passive`.
```

**Review by westonruter — CHANGES_REQUESTED** [review]
Thank you for adding these!

Should there also be notes added to the `wp-on`, `wp-on-window`, and `wp-on-document` sections to call out the async alternatives?

**Comment by westonruter** [comment]
See also https://github.com/WordPress/gutenberg/pull/62665 which would impact the docs here, especially the "async actions" section.

File: docs/reference-guides/interactivity-api/api-reference.md (hunk 1 above)
Comment (cbravobernal) [reply docs/reference-guides/interactivity-api/api-reference.md]: Done in https://github.com/WordPress/gutenberg/pull/62663/commits/e3b049796f2ef52635168d775e967d2cd7cb338f

File: docs/reference-guides/interactivity-api/api-reference.md (hunk 2 above)
Comment (cbravobernal) [reply docs/reference-guides/interactivity-api/api-reference.md]: Done in https://github.com/WordPress/gutenberg/pull/62663/commits/e3b049796f2ef52635168d775e967d2cd7cb338f

File: docs/reference-guides/interactivity-api/api-reference.md (hunk 3 above)
Comment (cbravobernal) [reply docs/reference-guides/interactivity-api/api-reference.md]: Done in https://github.com/WordPress/gutenberg/pull/62663/commits/e3b049796f2ef52635168d775e967d2cd7cb338f

**Review by westonruter — APPROVED** [review]

**Comment by ellatrix** [comment]
I just cherry-picked this PR to the wp/6.6-rc-1 branch to get it included in the next release: c8372ed56a

**Pull request merged**